utils/httpUtils.ts

```diff
@@ -72,3 +71,114 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlagsFromRequest = context => {
+  return context.request.headers.get("x-should-read-from-nextgen");
+};
+
+export const shouldReadFromNextGen = async context => {
+  let shouldReadFromNextGen = getFeatureFlagsFromRequest(context);
+  if (shouldReadFromNextGen === true || shouldReadFromNextGen === "true" || shouldReadFromNextGen === "True") {
+    // if the header is set, return the value
+    return true;
+  }
+  return false;
+};
+
+export const formatFedQueryForNextGen = (query: string) => {
+  let cleanQuery = query;
+  // remove fed Prefix
+  const firstCurlyBracket = query.indexOf("{");
```

Comment:
instead of this string manipulation, is it possible to just `JSON.parse()` the query and restructure it with actual objects?

Reply:
I was thinking about this but didn't go down the path because i didn't think the graphql could be `JSON.parse()` since it is not technically Javascript.